react/components/SKUSelector/components/SelectVariationMode.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+import styles from '../styles.css'
+import { DisplayOption } from '../types'
+import { Dropdown } from 'vtex.styleguide'
+import { injectIntl, InjectedIntlProps, defineMessages } from 'react-intl'
+
+interface VariationSelectModeProps extends InjectedIntlProps {
+  selectedItem: string | null
+  displayOptions: DisplayOption[]
+}
+
+const messages = defineMessages({
+  selectPlaceholder: {
+    id: 'store/sku-selector.select.placeholder',
+    defaultMessage: '',
+  }
+})
+
+function SelectVariationMode(props: VariationSelectModeProps) {
+  const {
+    intl,
+    selectedItem,
+    displayOptions,
+  } = props
+
+  const options = displayOptions.map(op => ({
+    label: op.label,
+    value: op.label,
+  }))
+
+  const mapOptions = () => 
+    displayOptions
+      .reduce<Record<string, DisplayOption>>((acc, cur) => {
+        acc[cur.label] = cur
+        return acc
+      }, {})
```

Comment:
What do you think of removing the function and moving this code inside `handleClick`? It's only used there.